config/opal_config_pthreads.m4

```diff
@@ -251,9 +252,10 @@ AC_PROVIDE_IFELSE([AC_PROG_CC],
                   [OPAL_INTL_POSIX_THREADS_PLAIN_C],
                   [opal_pthread_c_success=1])
 
-AC_PROVIDE_IFELSE([AC_PROG_CXX],
-                  [OPAL_INTL_POSIX_THREADS_PLAIN_CXX],
-                  [opal_pthread_cxx_success=1])
+AS_IF([test "$WANT_MPI_CXX_SUPPORT" = 1],
```

Comment:
You use `-eq` for `$WANT_MPI_CXX_SUPPORT` above -- should probably use it here, too.  Then you don't need to use the quotes for `$WANT_MPI_CXX_SUPPORT, either (because it's guaranteed to be a number).